SplunkRumWorkspace/SplunkRum/SplunkRum/DiskCache/CoreDataManager.swift

```diff
@@ -0,0 +1,279 @@
+            span.setValue(String(describing: pendingSpan.parentSpanId), forKey: "parentSpanId")
+            span.setValue(String(describing: pendingSpan.spanId), forKey: "spanId")
+            span.setValue(pendingSpan.name, forKey: "spanName")
+            span.setValue(pendingSpan.startTime, forKey: "start")
+            span.setValue(String(describing: pendingSpan.traceFlags), forKey: "traceFlags")
+            span.setValue(String(describing: pendingSpan.traceId), forKey: "traceId")
+            span.setValue(String(describing: pendingSpan.traceState), forKey: "traceState")
+            span.setValue(Date(), forKey: TimeStampColumn)  // this is used only for flush out
+
+            // save in to db
+            do {
+                try managedObjectContext.save()
+            } catch let error as NSError {
+                print("could not save. \(error) \(error.userInfo)")
+            }
+        }
+    }
+    /** Fetch span information from database and generate new span from it */
+    public func fetchSpanValues() -> [SpanData] {
+        let managedObjectContext = persistentContainer.viewContext
+        let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: ENTITY_NAME)
+        var newSpans = [SpanData]()
+        do {
+            let spans = try managedObjectContext.fetch(fetchRequest) as! [NSManagedObject]
+            for span in spans as [NSManagedObject] {
+                newSpans.append(createNewSpan(using: span))
+            }
+        } catch {
+            print(error)
+        }
+        return newSpans
+   }
+
+    func createNewSpan(using spanInfo: NSManagedObject) -> SpanData {
+        let tracer = buildTracer()
+        // set span kind logic here
+         let kind = spanInfo.value(forKey: "kind") as! String
+         let span = tracer.spanBuilder(spanName: spanInfo.value(forKey: "spanName") as! String).setSpanKind(spanKind: SpanKind(rawValue: kind)!).setStartTime(time: spanInfo.value(forKey: "start") as! Date).startSpan()
+        span.addEvent(name: spanInfo.value(forKey: "events") as! String)
+        let attributesDict = convertStringToDictionary(text: spanInfo.value(forKey: "attributes") as! String)
+        for dict in attributesDict! {
+            let value = dict.value.allValues[0]
+            if value is String {
+                span.setAttribute(key: dict.key, value: value as! String)
+            } else if dict.value is Int {
+                span.setAttribute(key: dict.key, value: value as! Int)
+            } else if dict.value is Double {
+                span.setAttribute(key: dict.key, value: value as! Double)
+            } else if dict.value is Bool {
+                span.setAttribute(key: dict.key, value: value as! Bool)
+            } else if dict.value is [String: Any] {
+                span.setAttribute(key: dict.key, value: value as? AttributeValue)
+            }
+        }
+        span.end(time: spanInfo.value(forKey: "duration") as! Date)
```

Comment:
Is the `SpanContext` lost here?

Reply:
@seemk, Here we are creating custom span using information stored in database, when we create custom span I think we are not using SpanContext.

Reply:
@seemk We can create SpanContext using traceId, spanID,t raceFlags, traceState. As mention in my very first  comment traceId, spanId, traceFlags are private property, So we can’t able to access it and create span context using that information stored on disk.